cartography/intel/azure/sql.py

```diff
@@ -277,267 +291,237 @@ def load_server_details(neo4j_session, credentials, subscription_id, details, up
     """
 
-    for data in ad_admins:
-        neo4j_session.run(
-            ingest_ad_admins,
-            AdAdminId=data['id'],
-            Name=data['name'],
-            AdministratorType=data['administrator_type'],
-            Login=data['login'],
-            ServerId=data['server_id'],
-            azure_update_tag=update_tag,
-        )
+    neo4j_session.run(
+        ingest_ad_admins,
+        ad_admins_list=ad_admins,
+        azure_update_tag=update_tag,
+    )
 
 
 @timeit
-def _load_recoverable_databases(neo4j_session, recoverable_databases, update_tag):
+def _load_recoverable_databases(
+        neo4j_session: neo4j.Session, recoverable_databases: List[Dict], update_tag: int
+) -> None:
     """
     Ingest the recoverable database details into neo4j.
     """
     ingest_recoverable_databases = """
-    MERGE (rd:AzureRecoverableDatabase{id: {DatabaseId}})
+    UNWIND {recoverable_databases_list} as rec_dbs
+    MERGE (rd:AzureRecoverableDatabase{id: rec_dbs.id})
     ON CREATE SET rd.firstseen = timestamp(), rd.lastupdated = {azure_update_tag}
-    SET rd.name = {Name},
-    rd.edition = {Edition},
-    rd.servicelevelobjective = {ServiceLevelObjective},
-    rd.lastbackupdate = {BackupDate}
-    WITH rd
-    MATCH (s:AzureServer{id: {ServerId}})
+    SET rd.name = rec_dbs.name,
+    rd.edition = rec_dbs.edition,
+    rd.servicelevelobjective = rec_dbs.service_level_objective,
+    rd.lastbackupdate = rec_dbs.last_available_backup_date
+    WITH rd, rec_dbs
+    MATCH (s:AzureSQLServer{id: rec_dbs.server_id})
     MERGE (s)-[r:RESOURCE]->(rd)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {azure_update_tag}
     """
 
-    for data in recoverable_databases:
-        neo4j_session.run(
-            ingest_recoverable_databases,
-            DatabaseId=data['id'],
-            Name=data['name'],
-            Edition=data['edition'],
-            ServiceLevelObjective=data['service_level_objective'],
-            BackupDate=data['last_available_backup_date'],
-            ServerId=data['server_id'],
-            azure_update_tag=update_tag,
-        )
+    neo4j_session.run(
+        ingest_recoverable_databases,
+        recoverable_databases_list=recoverable_databases,
+        azure_update_tag=update_tag,
+    )
 
 
 @timeit
-def _load_restorable_dropped_databases(neo4j_session, restorable_dropped_databases, update_tag):
+def _load_restorable_dropped_databases(
+        neo4j_session: neo4j.Session, restorable_dropped_databases: List[Dict], update_tag: int
+) -> None:
     """
     Ingest the restorable dropped database details into neo4j.
     """
     ingest_restorable_dropped_databases = """
-    MERGE (rdd:AzureRestorableDroppedDatabase{id: {DatabaseId}})
+    UNWIND {restorable_dropped_databases_list} as res_dropped_dbs
+    MERGE (rdd:AzureRestorableDroppedDatabase{id: res_dropped_dbs.id})
     ON CREATE SET rdd.firstseen = timestamp(), rdd.lastupdated = {azure_update_tag}
```

Comment:
lastupdated should not be in the create set. it should be in update set.

Reply:
done